osc/git_scm/store.py

```diff
@@ -29,27 +29,32 @@ def is_package_dir(cls, path):
     def __init__(self, path, check=True):
         self.path = path
         self.abspath = os.path.abspath(self.path)
-        try:
-            self.toplevel = self._run_git(["rev-parse", "--show-toplevel"])
-            self.toplevel = os.path.abspath(self.toplevel)
-        except subprocess.CalledProcessError:
-            self.toplevel = None
 
-        # TODO: how to determine if the current git repo contains a project or a package?
         self.is_project = False
         self.is_package = False
 
-        if self.toplevel:
+        self.project_obs_scm_store = None
+
+        parent_dir = os.path.join(self.abspath, "..")
```

Comment:
ideally we would not only look on directory, because we have git repositories using subdir mechanic already. The config file will change, but the fact that package sources may exist in .../project_level/some/sub/directory/package   exists. IIRC git is searching upwards until / looking for .git directory. Maybe we can do the same here?

Reply:
This basically means we'll be executing `git` every time osc is ran outside a workdir without `.osc` in it.
Couldn't we narrow down definition of a package? If we knew that it lives in a certain location such as `<project-dir>/packages/` or if it's easily identified with metadata, we could build smarter logic around this lookup.

To avoid calling `.git`, I could implement the traversal in osc, but it somehow doesn't seem entirely right to me.